linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faildelay_delay/ansible/ubuntu.yml

```diff
@@ -0,0 +1,23 @@
+# platform = multi_platform_ubuntu
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+- name: Ensure pam_faildelay.so is properly configured in /etc/pam.d/common-auth
+  block:
+    - name: Check existent entry and get delay
+      shell: |
+        set -o pipefail
+        grep "required pam_faildelay.so" /etc/pam.d/common-auth
+        sed 's/.*\= *//'
+      register: pam_delay
+      changed_when: false
+    
+    - name: Set proper entry
```

Comment:
This task should be enough. `lineinfile` module will ensure the desired line is correct and won't change it if already correct. So, you don't need to collect the delay value before and test it, unless you want to accept values higher than `4000000`. I think it would be worth to simplify it and avoid using `shell` module. Alternatively you can use `lineinfile` in check mode and `set_fact` modules instead of using `shell`.

Reply:
Thank you for this feedback, I'll remove this and leave the remediation with this task only